manim/mobject/types/vectorized_mobject.py

```diff
@@ -1023,6 +1049,50 @@ def point_from_proportion(self, alpha: float) -> np.ndarray:
 
             current_length += length
 
+    def proportion_from_point(
+        self,
+        point: typing.Iterable[typing.Union[float, int]],
+    ) -> float:
+        """Returns the proportion along the path of the :class:`VMobject`
+        a particular given point is at.
+
+        Parameters
+        ----------
+        point (typing.Iterable[float])
+            The Cartesian coordinates of the point which may or may not lie on the :class:`VMobject`
+
+        Returns
+        -------
+        float
+            The proportion along the path of the :class:`VMobject`.
+
+        Raises
+        ------
+        :exc:`ValueError`
+            If ``point`` does not lie on the curve.
+        :exc:`Exception`
+            If the :class:`VMobject` has no points.
+        """
+        self.throw_error_if_no_points()
+
+        nc = self.get_num_curves()
+        vmob_length = sum(self.get_nth_curve_length(i) for i in range(nc))
+        total_length = 0
+        for n in range(nc):
+            control_points = self.get_nth_curve_points(n)
+            length = self.get_nth_curve_length(n)
+            if point_lies_on_bezier(point, control_points):
+                t = max(bez_params_from_point(point, control_points))
+                total_length += length * t
+                break
+            else:
+                total_length += length
```

Comment:
The `break` means the code below won't get executed anyways, making the `else` redundant
```suggestion

            if point_lies_on_bezier(point, control_points):
                t = max(bez_params_from_point(point, control_points))
                total_length += length * t
                break

            total_length += length
```

Reply:
Done.